typescript-service-generator-core/src/main/java/com/palantir/code/ts/generator/ServiceClassParser.java

```diff
@@ -60,26 +61,35 @@
     }
 
     public ServiceModel parseServiceClass(Class<?> serviceClass, TypescriptServiceGeneratorConfiguration settings) {
+        return this.parseServiceClass(Lists.newArrayList(serviceClass), settings);
+    }
+
+    public ServiceModel parseServiceClass(List<Class<?>> serviceClasses, TypescriptServiceGeneratorConfiguration settings) {
         ImmutableServiceModel.Builder serviceModel = ImmutableServiceModel.builder();
-        Path servicePathAnnotation = serviceClass.getAnnotation(Path.class);
-        serviceModel.servicePath(PathUtils.trimSlashes(servicePathAnnotation.value()));
-        serviceModel.name(serviceClass.getSimpleName());
-
-        Set<Method> serviceMethods = getAllServiceMethods(serviceClass);
-        // find and stores all types that are referenced by this service
-        Set<Type> referencedTypes = Sets.newHashSet();
-        for (Method method : serviceMethods) {
-            referencedTypes.addAll(getTypesFromEndpoint(method, settings));
-        }
-        serviceModel.referencedTypes(referencedTypes);
+        serviceModel.name(serviceClasses.get(0).getSimpleName());
```

Comment:
This behavior is a bit subtle -- nothing about the signature implies to me that order matters.